### PR TITLE
Update dg branch

### PR DESCRIPTION
**Pull request by JrmCkh**
1. Update the pUML in Design portion
2. Updated the ordering of DG

### CODE REVIEW

File: docs/diagrams/LogicClassDiagram.puml [hunk 1]
```diff
@@ -4,31 +4,28 @@ skinparam arrowThickness 1.1
...
 Class CommandResult
 Class "{abstract}\nCommand" as Command
 
-
 Class "<<interface>>\nLogic" as Logic
 Class LogicManager
 }
 
-package Model{
+Package Model{
```
Comment (lhy-hoyin) [inline docs/diagrams/LogicClassDiagram.puml:17]: will changing the capital letter affect anything?

File: docs/diagrams/StorageClassDiagram.puml [hunk 2]
```diff
@@ -4,7 +4,7 @@ skinparam arrowThickness 1.1
 skinparam arrowColor STORAGE_COLOR
 skinparam classBackgroundColor STORAGE_COLOR
 
-package Storage{
+package Storage as StoragePackage{
```
Comment (lhy-hoyin) [inline docs/diagrams/StorageClassDiagram.puml:7]: How does this affect?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -2,31 +2,76 @@
 layout: page
 title: Developer Guide
 ---
-* Table of Contents
-{:toc}
-
```
Comment (lhy-hoyin) [inline docs/DeveloperGuide.md:7]: I guess it was originally auto-generated. Have you checked if we can still rely on the auto-generated one?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -51,34 +97,36 @@ The rest of the App consists of four components.
...
+the command `delete 1`.
 
 <img src="images/ArchitectureSequenceDiagram.png" width="574" />
 
 Each of the four main components (also shown in the diagram above),
 
 * defines its *API* in an `interface` with the same name as the Component.
-* implements its functionality using a concrete `{Component Name}Manager` class (which follows the corresponding API `interface` mentioned in the previous point.
+* implements its functionality using a concrete `{Component Name}Manager` class (which follows the corresponding 
+  API `interface` mentioned in the previous point.
```
Comment (lhy-hoyin) [inline docs/DeveloperGuide.md:111]: hmmmm....is this missing a closing bracket?
(change requested here)

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -157,183 +198,49 @@ The `Storage` component,
...
 
+<h2 id="documentation">Documentation, Logging, Testing, Configuration, Dev-Ops</h2>
 --------------------------------------------------------------------------------------------------------------------
-
-## **Documentation, logging, testing, configuration, dev-ops**
 * [Documentation guide](Documentation.md)
 * [Testing guide](Testing.md)
 * [Logging guide](Logging.md)
 * [Configuration guide](Configuration.md)
 * [DevOps guide](DevOps.md)
```
Comment (lhy-hoyin) [inline docs/DeveloperGuide.md:235]: This looks like its referencing to different files. Will this be an issue when we try to convert the DG to PDF?

**Review by lhy-hoyin — CHANGES_REQUESTED** [review]

File: docs/diagrams/LogicClassDiagram.puml (hunk 1 above)
Comment (JrmCkh) [reply docs/diagrams/LogicClassDiagram.puml:17]: Nope. Intended for standardization

File: docs/diagrams/StorageClassDiagram.puml (hunk 2 above)
Comment (JrmCkh) [reply docs/diagrams/StorageClassDiagram.puml:7]: The code blow has a class named Storage. Without using the 'as ...', the other arrows that which are meant to refer to the 'Storage Class' ends up referring to this 'Storage Package'. Leading to the situation below.

image

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (JrmCkh) [reply docs/DeveloperGuide.md:111]: Good catch. This issue exists from AB3 itself.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (JrmCkh) [reply docs/DeveloperGuide.md:235]: This was copied over from AB3. If there is an issue when converting to PDF, perhaps we could simply copy and paste over?

File: docs/diagrams/LogicClassDiagram.puml (hunk 1 above)
Comment (JrmCkh) [reply docs/diagrams/LogicClassDiagram.puml:17]: Will be converting back to small caps, as it seems to be the more frequently used.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (JrmCkh) [reply docs/DeveloperGuide.md]: reverted back. AB3 uses a plugin of some sort called "markdown kramdown" for its toc function.

**Review by lhy-hoyin — APPROVED** [review]
LGTM

**Pull request merged**